krisHutchison/answers.md

```diff
@@ -0,0 +1,240 @@
+
+
+**A:** True - "0"+1 evaluates to string "01" which is converted to a number when compared to number 1. The leading 0 on the string "01" is dropped.
+
+**p)** `(typeof (x+1))===(typeof x)`	
+
+
+**A:** Error - (x+1) is evaluated first but x is undefined so the operation terminates.
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+
+
+**A:** Error - the expression is evaluated from within the inner parens. Typeof x is undefined which returns true when evaluated against not string 'number' the operation terminates when it gets to the undefined x on the left side of the expression.
+
+**r)** `(x=(typeof (x+(typeof x))))==x`
+
+
+**A:** Error - the inner (typeof x) evaluates to "undefined" which is added to x leaving a string 'xundefined' which is typeof string but the first x terminates because it's undefined.
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+**2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+
+
+**A:** 
+`(x+y+z)/3`
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+
+**A:** 
+
+	x = x-(((x-(x+y+z)/3))*0.5)
+	y = ((((x+y+z)/3)-y)*0.5)+y
+    z = ((((x+y+z)/3)-z)*0.5)+z
+
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+
+
+**A:** 
+`(r-l) * (t-b)`
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+
+**A:** 
+`(t-b) > (r-l)`
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+
+
+**A:** 
+`(t-b) > (r-l) ? (r-l) * 3.14 : (t-b) * 3.14`
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+
+**A:** 
+`(Math.sqrt((Math.pow((t-b), 2) + Math.pow((r-l), 2))) *3.14);`
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+
+**A:** 
+
+	leftHorizontal = ((r-l) / 3 ) - l  
+	rightHorizontal = ((r-l) /3 ) - r
+	bottomVertical = ((t-b) / 3 ) -b
+	topVertical = ((t-b) / 3 ) -t 
+
```

Comment:
If you imagine a square with the bottom left corner on (0,0) and a side length of 3, you would end up with your  both your horizontals and your verticals at -1 and 2, essentially shifting the original square one unit down and to the left.